### DISCUSSION TITLE
PETSC error appearing after coloring

### DISCUSSION THREAD

**Discussion by antoineblgipsa** [mdolab/dafoam#661] [Bug] [answered]
Hello everyone.

I am trying to make my first simulation using DAFoam. However, I am meeting some issues after the first coloring steps. On the first screenshot, we can see that there is a first error occuring about "addGeoDVLocal" (addGeoDVLocal will be deprecated, use addLocalDV instead). I tried to update to pygeo 1.13.1 as mentionned on the following discussion (Compile from source problem : https://github.com/mdolab/dafoam/discussions/309).I have obtained the same result and I have finally solved that part by replacing with addLocalDV in my script. 

However, I have still an issue mentionning PETSC and the following error : (UnicodeDecodeError : 'utf-8' codec can't decode byte 0x89 in position 48 : invalid start byte) at the bottom. Can you help me with that bug ?

1st screenshot
![image](https://github.com/mdolab/dafoam/assets/80889173/06ee2359-6ca4-407d-9a48-bf44f1fc1247)

2nd screenshot
![image](https://github.com/mdolab/dafoam/assets/80889173/77655e2a-77a4-4015-a188-27644015c91d)

**Comment by friedenhe** [comment]
Which case are you running? and which DAFoam version are you using?

**Reply by antoineblgipsa** [reply]
Hello Mr Ping He. 

Thank you for your answer.

The script that I am running is similar to the one used in the example of the Omera M6 Wing but without CL objective (only drag) applied to a spheric object. I am also using DAFoam v3.0.0. 

The anomaly with addGeoDVLocal doesn't seem to generate problems on my previous simulations. 

I suspect a problem with the installation of petsc4py (with my virtual machine). The PETSC error only seems to appear when I am working with a virtual machine. When I try do the same simulation locally (on my PC with a coarser mesh) everything seems to work perfectly fine (the simulation goes on with no petsc error). I cannot explain why.

**Comment by friedenhe** [comment]
The addGeoDVLocal is an old API. Please use the latest DAFoam version and the latest tutorials. We never test the code on a virtual machine, so the Petsc error is probably related to the virtual machine environment.

**Reply by antoineblgipsa** [reply]
Hello Sir,

As you suggested, I tried different versions of DAFoam (including the latest) and unfortunately it has not solved my problem. However, I have different kind of errors depending on the version. With DAFoam 3.0.0, I have the PETSC error that I have already shown (1). With previous releases (2.2.10 and 2.2.9), my simulation is freezed right after the coloring (2). I can wait several hours without any change on my screen. Finally, on the latest version, I have an error mentioning "TensorFlow" (3).

Image (1)
![(1)](https://github.com/user-attachments/assets/3257bee7-b2e7-44d4-b0e8-8907e08e4118)

Image (2)
![(2)](https://github.com/user-attachments/assets/6f05b32e-bed6-4ccb-8d12-a44d7cfe459a)

Image (3)
![(3)](https://github.com/user-attachments/assets/0a5aee40-c775-44d9-a6e1-94d799da196a)

For the third image 
![image](https://github.com/user-attachments/assets/91cdfc09-0ccc-4da7-b9f5-f43fae57db4e)



I tried to obtain answers on other forums but I didn't find any post or questions involving the same situation unfortunately. Have a good day.

**Reply by antoineblgipsa** [reply, marked as answer]
Hello Sir.

It seems that my team has successfully solved the problem on our virtual machines. For some reasons, I was missing a script indicating the standard procedure for our simulations. The problem was not linked to DAFoam fortunately. Thank you very much for your help and your amazing software.

Have a good day.
